src/components/input/SelectNext.tsx

```diff
@@ -287,7 +287,7 @@ function SelectMain<T>({
   const buttonRef = useSyncedRef(elementRef);
   const defaultButtonId = useId();
   const extraProps = useMemo(
-    () => (listboxAsPopover ? { popover: '' } : {}),
+    () => (listboxAsPopover ? { popover: 'auto' } : {}),
```

Comment:
The typecheck still fails, as it detects `auto` as string. One option is setting `'auto' as const`, but since now it is part of the type, you can also move this logic to a regular prop where the value is dynamically set to `auto` or `false`.

Reply:
🤦 - Indeed. I ran the tests locally but forgot to run typechecking. I moved `popover` to be an unconditionally set prop, but had to use the value `undefined` rather than `false` for the case where `listboxAsPopover` is disabled, since Preact is not handling boolean values correctly for this prop.